### PR TITLE
[PROCS-1464] Add Process Discovery Feature to Operator

### PR DESCRIPTION
**Pull request by just-chillin**
### What does this PR do?

This PR adds Process Discovery as a configurable feature in the datadog operator.

### Motivation

Process discovery is a feature that has been in the Agent for several releases, and is already in the helm chart.

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: 7.32
* Cluster Agent: N/A

### Describe your test plan

Deploy the operator in kind with container collection disabled and make sure the process agent starts up.

### Checklist

- [x] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [x] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

File: docs/configuration.v2alpha1.md [hunk 1]
```diff
@@ -114,6 +114,7 @@ spec:
 | features.otlp.receiver.protocols.grpc.endpoint | Endpoint for OTLP/gRPC. gRPC supports several naming schemes: https://github.com/grpc/grpc/blob/master/doc/naming.md The Datadog Operator supports only 'host:port' (usually `0.0.0.0:port`). Default: `0.0.0.0:4317`. |
 | features.otlp.receiver.protocols.http.enabled | Enable the OTLP/HTTP endpoint. |
 | features.otlp.receiver.protocols.http.endpoint | Endpoint for OTLP/HTTP. Default: '0.0.0.0:4318'. |
+| features.processDiscoveryCollection.enabled | Enables process discovery collection Default: true |
```
Comment (robertjli) [inline docs/configuration.v2alpha1.md:117]:
```suggestion
| features.processDiscoveryCollection.enabled | Enables process discovery collection. Default: true |
```

**Review by robertjli — APPROVED** [review]

File: docs/configuration.v2alpha1.md (hunk 1 above)
Comment (just-chillin) [reply docs/configuration.v2alpha1.md]: Fixed!

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 2]
```diff
@@ -196,6 +198,15 @@ type LiveContainerCollectionFeatureConfig struct {
 	Enabled *bool `json:"enabled,omitempty"`
 }
 
+// ProcessDiscoveryFeatureConfig contains the configuration for the process discovery check
+// ProcessDiscovery is run in the ProcessAgent
+type ProcessDiscoveryFeatureConfig struct {
+	// Enables process discovery collection.
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:204]:
```suggestion
	// Enabled enables process discovery collection.
```

File: controllers/datadogagent/feature/ids.go [hunk 3]
```diff
@@ -23,6 +23,8 @@ const (
 	LiveContainerIDType = "live_container"
 	// LiveProcessIDType Live Process feature.
 	LiveProcessIDType = "live_process"
+	// ProcessDiscoveryIdType Process Discovery feature.
+	ProcessDiscoveryIdType = "process_discovery"
```
Comment (celenechang) [inline controllers/datadogagent/feature/ids.go:27]:
```suggestion
	ProcessDiscoveryIDType = "process_discovery"
```
maybe eventually we'll switch to `Id` 😅 , but for consistency's sake

File: controllers/datadogagent/feature/processdiscovery/feature.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+func buildProcessDiscoveryFeature(options *feature.Options) feature.Feature {
+	return &processDiscoveryFeature{}
+}
+
+type processDiscoveryFeature struct{}
+
+func (p processDiscoveryFeature) ID() feature.IDType {
+	return feature.ProcessDiscoveryIdType
```
Comment (celenechang) [inline controllers/datadogagent/feature/processdiscovery/feature.go:29]:
```suggestion
	return feature.ProcessDiscoveryIDType
```

**Review by celenechang — COMMENTED** [review]
looks good to me! just small nits

File: apis/datadoghq/v2alpha1/datadogagent_types.go (hunk 2 above)
Comment (just-chillin) [reply apis/datadoghq/v2alpha1/datadogagent_types.go:204]: Ah good suggestion updated!

**Review by celenechang — APPROVED** [review]

**Pull request merged**